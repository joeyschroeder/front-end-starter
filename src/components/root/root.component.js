import { ConnectedRouter } from 'connected-react-router';
import { Main } from '../main/main.component';
import PropTypes from 'prop-types';
import { Provider } from 'react-redux';
import { ROUTES } from '../../routes';
import React from 'react';
import { ReduxScreen } from '../../screens/redux/redux.screen';
import { Route } from 'react-router';
import { SplashScreen } from '../../screens/splash/splash.screen';

export const Root = props => {
    const { history, store } = props;

    return (
        <Provider store={store}>
            <ConnectedRouter history={history}>
                <Main>
                    <Route exact path={ROUTES.SPLASH} component={SplashScreen} />
                    <Route exact path={ROUTES.REDUX} component={ReduxScreen} />
                </Main>
            </ConnectedRouter>
        </Provider>
    );
};

Root.propTypes = {
    history: PropTypes.object,
    store: PropTypes.object
};

Root.defaultProps = {
    history: null,
    store: null
};
