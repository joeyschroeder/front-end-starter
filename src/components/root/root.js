import { Main } from 'components/main/main';
import PropTypes from 'prop-types';
import { Provider } from 'react-redux';
import { ROUTES } from 'constants/routes';
import React from 'react';
import { ReduxViewConnected } from 'components/redux-view/redux-view.connected';
import { Route, BrowserRouter } from 'react-router-dom';
import { Splash } from 'components/splash/splash';

export const Root = props => {
  const { store } = props;

  return (
    <Provider store={store}>
      <BrowserRouter>
        <Main>
          <Route exact path={ROUTES.SPLASH} component={Splash} />
          <Route exact path={ROUTES.REDUX} component={ReduxViewConnected} />
        </Main>
      </BrowserRouter>
    </Provider>
  );
};

Root.propTypes = {
  store: PropTypes.object
};

Root.defaultProps = {
  store: null
};
