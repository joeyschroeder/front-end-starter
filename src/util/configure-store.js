import { applyMiddleware, createStore } from 'redux';
import { connectRouter, routerMiddleware } from 'connected-react-router';

import Reducers from '../ducks/index';
import thunk from 'redux-thunk';

export const configureStore = history => {
    const store = createStore(
        connectRouter(history)(Reducers),
        applyMiddleware(routerMiddleware(history), thunk)
    );

    if (module.hot) {
        module.hot.accept('../ducks/index', () => {
            const nextReducers = require('../ducks/index');
            store.replaceReducer(nextReducers);
        });
    }

    return store;
};
