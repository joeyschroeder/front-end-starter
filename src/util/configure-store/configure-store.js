import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { createRootReducer } from 'store/create-root-reducer';

export function configureStore() {
  const store = createStore(createRootReducer(), applyMiddleware(thunk));

  return store;
}
