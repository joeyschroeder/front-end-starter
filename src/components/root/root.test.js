import { Root } from './root';
import ShallowRenderer from 'react-test-renderer/shallow';
import { configureStore } from 'util/configure-store/configure-store';
import { createMemoryHistory } from 'history';
/* global React */

describe('<Root />', () => {
  const renderer = new ShallowRenderer();

  // keyLength is set to zero so a new key is not
  // generated during every snapshot comparison
  const history = createMemoryHistory({ keyLength: 0 });
  const store = configureStore(history);

  it('should render correctly', () => {
    renderer.render(<Root history={history} store={store} />);
    const result = renderer.getRenderOutput();

    expect(result).toMatchSnapshot();
  });
});
