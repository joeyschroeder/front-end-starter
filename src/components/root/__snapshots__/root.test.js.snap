// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Root /> should render correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <BrowserRouter>
    <Main>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(ReduxView)",
            "type": [Function],
          }
        }
        exact={true}
        path="/redux"
      />
    </Main>
  </BrowserRouter>
</Provider>
`;
